packages/node/src/indexer/indexer.manager.ts

```diff
@@ -151,19 +166,13 @@ export class IndexerManager {
             await this.poiService.getLatestPoiBlockHash(),
             this.project.id,
           );
-          if (this.blockOffset === undefined) {
-            await this.metadataRepo.upsert(
-              {
-                key: 'blockOffset',
-                value: blockHeight - 1,
-              },
-              { transaction: tx },
-            );
-            this.setBlockOffset(blockHeight - 1);
-          }
           poiBlockHash = poiBlock.hash;
           await this.storeService.setPoi(poiBlock, { transaction: tx });
           this.poiService.setLatestPoiBlockHash(poiBlockHash);
+          await this.storeService.setMetadataBatch(
```

Comment:
`lastPoiHeight` isn't the height that we have a poi record, but the height that we have mmrRoot calculated.

Reply:
lastPoiHeight is ok to be not realtime, so you won't do too many db writes when you are catching up mmr root calculation to poi table.